src/frontend/components/PictureInPictureLayer/usePIPDragger/utils.ts

```diff
@@ -0,0 +1,92 @@
+import { Nullable } from 'utils/types';
+
+export const defaultmarginSize = 40;
+
+export interface Point {
+  x: number;
+  y: number;
+}
+
+export const getVelocity = (point: Point) => {
+  return Math.sqrt(Math.pow(point.x, 2) + Math.pow(point.y, 2));
+};
+
+const anchorToClosestAngle = (
+  container: HTMLDivElement,
+  picture: HTMLDivElement,
+  defaultMargin: number,
+): Point => {
+  const pictureCenterPoint: Point = {
+    x: picture.offsetLeft + picture.offsetWidth / 2.0,
+    y: picture.offsetTop + picture.offsetHeight / 2.0,
+  };
+
+  if (
+    pictureCenterPoint.x < container.offsetWidth / 2.0 &&
+    pictureCenterPoint.y < container.offsetHeight / 2.0
+  ) {
+    return {
+      x: defaultMargin,
+      y: defaultMargin,
+    };
+  } else if (pictureCenterPoint.x < container.offsetWidth / 2.0) {
+    return {
+      x: defaultMargin,
+      y: container.offsetHeight - picture.offsetHeight - defaultMargin,
+    };
+  } else if (pictureCenterPoint.y < container.offsetHeight / 2.0) {
+    return {
+      x: container.offsetWidth - picture.offsetWidth - defaultMargin,
+      y: defaultMargin,
+    };
+  } else {
+    return {
+      x: container.offsetWidth - picture.offsetWidth - defaultMargin,
+      y: container.offsetHeight - picture.offsetHeight - defaultMargin,
+    };
+  }
+};
+
+const anchorToDistantAngle = (
+  container: HTMLDivElement,
+  picture: HTMLDivElement,
+  defaultMargin: number,
+  velocityDescription: Point,
+): Point => {
+  if (velocityDescription.x <= 0 && velocityDescription.y <= 0) {
+    return {
+      x: defaultMargin,
+      y: defaultMargin,
+    };
+  } else if (velocityDescription.x <= 0) {
+    return {
+      x: defaultMargin,
+      y: container.offsetHeight - picture.offsetHeight - defaultMargin,
+    };
+  } else if (velocityDescription.y <= 0) {
+    return {
+      x: container.offsetWidth - picture.offsetWidth - defaultMargin,
+      y: defaultMargin,
+    };
+  } else {
+    return {
+      x: container.offsetWidth - picture.offsetWidth - defaultMargin,
+      y: container.offsetHeight - picture.offsetHeight - defaultMargin,
+    };
+  }
+};
```

Comment:
I feel like `anchorToClosestAngle` and `anchorToDistantAngle` deserves some shared variable and functions.
Maybe something like 
```
if (inTopLeftArea(picture)) {
  return topLeftAngle
}
else(inTopRightArea(picture)) {
[…]
```
WDYT ?